base/src/CCNPPI.cpp

```diff
@@ -78,130 +896,55 @@ class CCNPPI::Detail
 				dstNextPtrOffset[i] = outputRawMetadata->getNextPtrOffset(i);
 			}
 		}
-		
-		return true;
-	}
 
-	bool compute(void* buffer, void* outBuffer)
-	{
-		for(auto i = 0; i < inputChannels; i++)
+		if (mIntermediate != nullptr)
 		{
-			src[i] = static_cast<const Npp8u*>(buffer) + srcNextPtrOffset[i];
-		}
-
-		for(auto i = 0; i < outputChannels; i++)
-		{
-			dst[i] = static_cast<Npp8u*>(outBuffer) + dstNextPtrOffset[i];
-		}
+			intermediateFrameType = mIntermediate->getFrameType();
 
-		if (inputImageType == ImageMetadata::YUV411_I)
-		{
-			lanuchAPPYUV411ToYUV444(src[0], srcPitch[0], dst, dstPitch[0], srcSize[0], props.stream);
-		}
-		else if (inputImageType == ImageMetadata::MONO && outputImageType == ImageMetadata::BGRA)
-		{
-			auto status = nppiDup_8u_C1C4R_Ctx(src[0],
-				srcPitch[0],
-				dst[0],
-				dstPitch[0],
-				srcSize[0],
-				nppStreamCtx
-			);
-			if (status != NPP_SUCCESS)
+			if (intermediateFrameType = FrameMetadata::RAW_IMAGE)
 			{
-				LOG_ERROR << "nppiCopy_8u_C1C4R_Ctx failed<" << status << ">";
-				return false;
-			}
+				auto outputRawMetadata = FrameMetadataFactory::downcast<RawImageMetadata>(mIntermediate);
```

Comment:
rename this variable please - this is so confusing

Reply:
changed it to intermediateRawMetadata